tests/store/artifact/test_s3_artifact_repo.py

```diff
@@ -362,6 +362,28 @@ def test_delete_artifacts(s3_artifact_repo, tmp_path):
     assert not tmpdir_objects
 
 
+def test_delete_artifacts_pagination(s3_artifact_repo, tmp_path):
+    subdir = tmp_path / "subdir"
+    subdir.mkdir()
+    subdir_path = str(subdir)
+    # The maximum number of objects that can be listed in a single call is 1000
+    # https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html
+    for i in range(1100):
+        with open(os.path.join(subdir_path, f"{i}.txt"), "w") as f:
```

Comment:
/ai `subdir_path` is a `pathlib.Path` object. Simplify this code using `write_text`

Reply:
Thanks, refactored the new ut and the existing one for `delete_artifacts`.